### PR TITLE
[hotfix][table-planner][table-api-java][docs] Move topn cache config option

### PR DESCRIPTION
**Pull request by matriv**
`table.exec.topn.cache-size` config option was previously defined inside
`StreamExecRank` and still marked as experimental. Move it to
`ExecutionConfigOptions` and generate the corresponding docs.

### CODE REVIEW

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java [hunk 1]
```diff
@@ -414,6 +415,16 @@
                             "Determines whether CAST will operate following the legacy behaviour "
                                     + "or the new one that introduces various fixes and improvements.");
 
+    @Documentation.TableOption(execMode = Documentation.ExecMode.STREAMING)
+    public static final ConfigOption<Long> TABLE_EXEC_TOPN_CACHE_SIZE =
+            ConfigOptions.key("table.exec.topn.cache-size")
```
Comment (twalthr) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java:420]: `topn` is an outdated name. Use `rank` everywhere like `table.exec.rank.cache-size` to be consistent. The old key can be declared as a deprecated key.

**Review by twalthr — CHANGES_REQUESTED** [review]
Thanks @matriv.

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java (hunk 1 above)
Comment (matriv) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java]: Thx, please check again!

**Comment by twalthr** [comment]
I know naming is a difficult issue but I would still prefer: `table.exec.rank.topn-cache-size` instead of `table.exec.rank-topn.cache-size`. Top-N belongs to the rank ExecNode and components of a key should be hierarchical.

**Comment by matriv** [comment]
> I know naming is a difficult issue but I would still prefer: `table.exec.rank.topn-cache-size` instead of `table.exec.rank-topn.cache-size`. Top-N belongs to the rank ExecNode and components of a key should be hierarchical.

No worries, I think I just didn't remember it, It's my fault, please check again.

**Review by twalthr — APPROVED** [review]

**Pull request closed without merging**